src/chapter7/breakfast.py

```diff
@@ -1,10 +1,11 @@
+from __future__ import print_function
```

Comment:
Hi @cclauss actually jython supports `print()` without the import from `__future__`, at least in 2.7

```
C:\Users\Adam\jython\jython4>dist\bin\jython.exe
Jython 2.7.2a1+ (, May 9 2019, 11:51:47)
[OpenJDK 64-Bit Server VM (Oracle Corporation)] on java11.0.1
Type "help", "copyright", "credits" or "license" for more information.
>>> print('hi')
hi
>>>
```
 
That should make the change less verbose. Thanks for taking the time to submit this and the other v3 compatibility PRs.

Reply:
Python porting best practice says that we should [__prevent compatibility regressions__](https://docs.python.org/3/howto/pyporting.html#prevent-compatibility-regressions).  This import ___changes the way that Python 2 operates___ so that __print__ statements are now syntax errors while __print()__ function continues to work as expected.  This prevents Python 2 developers from writing code that breaks compatibility with Python 3.

Proof:
* $ python2 -c "print 'This works...'"
* $ python2 -c "from __future__ import print_function ; print '...but this raises a syntax error.'"

Reply:
Thank you for explaining. I can see the systematic logic to this and understand why it is in the porting guidelines now. I think there is still a decision around its impact on inline explanations in a reference book, as Jeff points out.

While on the topic it is also unfortunately a jython bug with `__future__.print_function()` when using from an interactive console which might compound confusion.

https://bugs.jython.org/issue2007